tests/test_toolbox.py

```diff
@@ -601,6 +614,19 @@ def test_hypot(self):
             self.assertEqual(tb.hypot(numpy.array([3.,4.])), 5.0)
             self.assertEqual(tb.hypot(numpy.array([3,4])), 5.0)
 
+    def test_hypot_array(self):
+        """hypot when called on an array falls to C"""
+        invals = [ numpy.asarray([3, 4]), numpy.arange(3,6), numpy.arange(3,10), numpy.asarray([-1,2,3]) ]
+        ans = [ 5, 7.0710678118654755, 16.73320053068151, 3.7416573867739413 ]
+        for i, tst in enumerate(invals):
+            self.assertAlmostEqual(ans[i], tb.hypot(*tst))
+        for i, tst in enumerate(invals):
+            self.assertAlmostEqual(ans[i], tb.hypot(tst))
+        self.assertEqual(5.0, tb.hypot(5.0))
+        if spacepy.lib.have_libspacepy:
+            self.assertEqual(tb.hypot(numpy.array([3.,4.])), 5.0)
+            self.assertEqual(tb.hypot(numpy.array([3,4])), 5.0)
```

Comment:
It's not obvious what you're doing here, but presumably it's checking the fix with the require (maybe comment on that?). There should also probably be a test with a subclass (e.g. dmarray) and some tests with non-numpy inputs.

Reply:
This checks the `np.require` is doing its job and not passing the int thru to the C.